### ISSUE TITLE
PHP Fatal error:  Uncaught Symfony\Component\Debug\Exception\FatalThrowableError: Parse error: syntax error, unexpected 'const' (T_CONST), expecting variable (T_VARIABLE) in /vendor/doctrine/common/lib/Doctrine/Common/Proxy/AbstractProxyFactory.php:72

### ISSUE THREAD

**Issue by iraldoad** [doctrine/DoctrineBundle#818]
![doctrine-error](https://user-images.githubusercontent.com/15159678/40100139-e0c80608-58b0-11e8-9b2e-72a9920502e8.png)

**Comment by iraldoad** [comment]
Yes, I am running PHP7.0. This is what I have in my composer.json:

`"doctrine/doctrine-bundle": "^1.6"`
`"doctrine/orm": "^2.5"`

What should I do?

**Comment by alcaeus** [comment]
> What should I do?

Run `composer update` on a system running PHP 7.0 or follow the advice in [our blog post](https://www.doctrine-project.org/2017/07/25/php-7.1-requirement-and-composer.html), specifically the section "Making sure you get a compatible version".

**Comment by evanega3** [comment]
Hello, I have the same issue as you, I already put this on my composer file:


    "config": {
      "platform": {
          "php": "7.2.5"
      },

but I'm still having the same issue.

**Comment by xabbuh** [comment]
@evanega3 This setting will not magically make your PHP version behave as 7.2 if you are actually using an older version. You should rather use it when you have a more recent PHP version, but want to force the installation to assume that you have an older version (because your deployment target uses an older version, for example).

**Comment by evanega3** [comment]
@xabbuh, Hello, I already have 7.2.7 version on my live server.

**Comment by Ocramius** [comment]
@evanega3 sure it's not PHP CLI that is reporting that? The SAPI used by FPM or by the Apache2 integration are totally different binaries...

**Comment by evanega3** [comment]
@Ocramius, this is my new output:

PHP Fatal error:  Uncaught Symfony\\Component\\Debug\\Exception\\FatalThrowableError: Type error: Return value of Doctrine\\Commmon\\Annotations\\AnnotationRegistry::registerFile() must be an instance of Doctrine\\Common\\Annotations\\void, none returned in /var/www/testProject/vendor/doctrine/annotations/lib/Doctrine/Common/Annotations/AnnotationRegistry.php:67

Stack trace:

#0 /var/www/testProject/vendor/doctrine/annotations/lib/Doctrine/Common/Annotations/AnnotationReader.php(200): Doctrine\\Common\\Annotations\\AnnotationRegistry::registerFile('/var/www/testProj...')

#1 
/var/www/testProject/var/cache/dev/ContainerKsdfk89/appDevDebugProjectContainer.php(1187): Doctrine\\Common\\Annotations\\AnnotationReader->__construct()

#2 
/var/www/testProject/var/cache/dev/ContainerKsdfk89/appDevDebugProjectContainer.php(1177): ContainerKsdfk89\\appDevDebugProjectContainer->getAnnotations_ReaderService()

#3 
/var/www/testProject/var/cache/dev/ContainerKsdfk89/appDevDebugProjectContainer.php(2166): ContainerKsdfk89\\appDevDebugProjectContainer->getAnnotationReaderService()

#4 in /var/www/testProject/vendor/doctrine/annotations/lib/Doctrine/Common/Annotations/AnnotationRegistry.php on line 67

**Comment by Ocramius** [comment]
@evanega3 yes, that is most certainly your system running PHP 7.0 instead of 7.1.

**Comment by evanega3** [comment]
@Ocramius Hello, this is the output from the terminal console when I run php -v

PHP 7.2.7-1+ubuntu16.04.1+deb.sury.org+1 (cli) (built: Jun 22 2018 08:44:50) ( NTS )

Also from the composer.json:


"require": {
        "php": ">=7.1.1",
        ...
},
"config": {
      "platform": {
          "php": "7.2"
      },
      "sort-packages": true
    },

**Comment by Ocramius** [comment]
Your terminal != Your web server

**Comment by iraldoad** [comment]
Yes, it is very common to use different PHP configurations for the terminal and the web server.

**Comment by Ocramius** [comment]
Closing and locking here: upgrade your PHP installation.
